app/models/plan.rb

```diff
@@ -479,6 +479,19 @@ def clone_associations(custom)
 
   private
 
+  def scheduled_for_deletion?
+    # try(:account) because issuer could not be a service and could not respond to 'account'
+    !issuer || issuer.deleted? || ((account = issuer.try(:account)) && account.scheduled_for_deletion?)
```

Comment:
It think this comment is misleading, looking at the code, polymorphic relation `as: :issuer` is only set on Service and Account which all respond to `#account`

If the issuer is an `Account` then it is a `Provider`, so `issuer.account == Account.master`
If the issuer is a `Service` then `issuer.account` is a `Provider`

But all `Plan` respond_to `provider_account`

```suggestion
    !issuer || issuer.deleted? ||  provider_account&.scheduled_for_deletion?
```